.tito/lib/git_annex_spec_builder.py

```diff
@@ -0,0 +1,168 @@
+import os
+import shutil
+from distutils.version import LooseVersion as loose_version
+from pkg_resources import require
+
+from tito.compat import getstatusoutput
+from tito.config_object import ConfigObject
+from tito.builder import GitAnnexBuilder
+from tito.builder.main import BuilderBase
+from tito.common import error_out, debug, run_command, get_spec_version_and_release, find_spec_like_file, \
+                        get_relative_project_dir, get_relative_project_dir_cwd
+
+
+class GitAnnexSpecBuilder(GitAnnexBuilder):
+
+    def __init__(self, name=None, tag=None, build_dir=None,
+            config=None, user_config=None,
+            args=None, **kwargs):
+
+        """
+        name - Package name that is being built.
+
+        version - Version and release being built.
+
+        tag - The git tag being built.
+
+        build_dir - Temporary build directory where we can safely work.
+
+        config - Merged configuration. (global plus package specific)
+
+        user_config - User configuration from ~/.titorc.
+
+        args - Optional arguments specific to each builder. Can be passed
+        in explicitly by user on the CLI, or via a release target config
+        entry. Only for things which vary on invocations of the builder,
+        avoid using these if possible.  *Given in the format of a dictionary
+        of lists.*
+        """
+        ConfigObject.__init__(self, config=config)
+        BuilderBase.__init__(self, name=name, build_dir=build_dir, config=config,
```

Comment:
I must not be doing it right. How would I call super?

Reply:
Looking at the source of tito this looks a lot like a copy of Builder.__init__ with some slight differences: how ```self.git_commit_id``` and ```self.relative_project_dir``` are determined. Makes me wonder why that is and a bit hesitant to change it. Just calling super will mean duplicate work and potential issues. IMHO the inheritance used isn't well suited for extentension. Let's leave the current code as-is since it works.